fog/ingest/client/tests/get_ingress_key_records.rs

```diff
@@ -0,0 +1,265 @@
+            )))
+            .stdout(predicate::str::contains(format!(
+                "{}",
+                hex::encode(&ingest_server_set_up_data.retired_ingress_pubkey.get_data())
+            )))
+            .stdout(
+                predicate::str::contains(format!(
+                    "{}",
+                    hex::encode(&ingest_server_set_up_data.lost_ingress_pubkey.get_data())
+                ))
+                .not(),
+            );
+    }
+
+    // Test that the "--include-lost" and "--include-retired" options
+    // include active, retired, and lost keys.
+    {
+        let mut cmd = Command::cargo_bin("fog_ingest_client").unwrap();
+        cmd.arg("--uri")
+            .arg(&ingest_server_set_up_data.ingest_server_client_uri)
+            .arg("get-ingress-public-key-records")
+            .arg("--start-block-at-least")
+            .arg("0")
+            .arg("--include-lost")
+            .arg("--include-retired")
+            .assert()
+            .success()
+            .stdout(predicate::str::contains(format!(
+                "{}",
+                hex::encode(&ingest_server_set_up_data.active_ingress_pubkey.get_data())
+            )))
+            .stdout(predicate::str::contains(format!(
+                "{}",
+                hex::encode(&ingest_server_set_up_data.retired_ingress_pubkey.get_data())
+            )))
+            .stdout(predicate::str::contains(format!(
+                "{}",
+                hex::encode(&ingest_server_set_up_data.lost_ingress_pubkey.get_data())
+            )));
+    }
+    // Test that the "--start-block-at-least" option works correctly.
+    // This should exclude the retired ingress key because its start block is
+    // 123.
+    {
+        let mut cmd = Command::cargo_bin("fog_ingest_client").unwrap();
+        cmd.arg("--uri")
+            .arg(&ingest_server_set_up_data.ingest_server_client_uri)
+            .arg("get-ingress-public-key-records")
+            .arg("--start-block-at-least")
+            .arg("200")
+            .arg("--include-lost")
+            .arg("--include-retired")
+            .assert()
+            .success()
+            .stdout(predicate::str::contains(format!(
+                "{}",
+                hex::encode(&ingest_server_set_up_data.active_ingress_pubkey.get_data())
+            )))
+            .stdout(
+                predicate::str::contains(format!(
+                    "{}",
+                    hex::encode(&ingest_server_set_up_data.retired_ingress_pubkey.get_data())
+                ))
+                .not(),
+            )
+            .stdout(predicate::str::contains(format!(
+                "{}",
+                hex::encode(&ingest_server_set_up_data.lost_ingress_pubkey.get_data())
+            )));
+    }
+}
+
+/// Contains data pertaining to the set up of the ingest server and the ingress
+/// public keys.
+struct IngestServerSetUpData {
```

Comment:
Can we share this object with the object from the `sync-keys-from-remote` test?

Reply:
I think this could be done like this:

https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests

or we could add it to a `test-infra` crate i guess, but the above is likely preferable?

Reply:
Discussed offline- since the SetUpData differs between tests and would require us to pass params that aren't needed by either test (thereby modifying the set up method), we won't create a shared struct.